docs/developer/extending/apps/architecture/apl.mdx

```diff
@@ -1,3 +1,105 @@
 ---
 title: APL - Auth Persistence Layer
 ---
+
+APL is the core component of communication between the Saleor and the App.
+
+During installation, Saleor sends AppToken, which the App uses to communicate with graphQL API. The App must save this token and target domain - but where?
+
+# What is APL
+
+APL is an abstraction on key/value persistence that holds domains (under which Saleor API is available) and tokens for communication.
+
+Saleor provides a simple persistence with App Metadata. Still, if App doesn't have a token and domain saved (which is typical for serverless architecture), it can't also call Saleor to reach Metadata. Hence need for simple, fast storage to save auth data.
+
+Saleor doesn't require App to use any specific technology for persistence. That's why there are many implementations of APL available in the app-sdk package. And writing a custom one is easy.
+
+# Communication diagram
+
+This diagram explains flow how APL is used:
+
+Following diagram describes how APL is used to store token received from Saleor during registration
+
+sequenceDiagram
+Saleor->>+App: Call /api/register with Auth Token and Domain header
+App->>+APL: Save token and header
+APL->>+External Database: Save in persistence layer
+
+Follow diagram describes example of external webhook, that calls the App and should send data to Saleor
+
+sequenceDiagram
+External Service->>+App: Call endpoint with webhook and Saleor domain
+App->>+APL: Get token for received domain
+APL->>+External Database: Ask for token in persistence layer
+External Database->>+APL: Provide saved token
+APL->>+App: Provide saved token
+App->>+Saleor: Call graphQL query for related domain with retrieved token
```

Comment:
![Zrzut ekranu 2022-10-24 o 17 08 21](https://user-images.githubusercontent.com/9268745/197560513-1282156b-3a00-4fa7-97ca-721139dfd76e.png)

Reply:
@peelar can we style mermaid rendered a little bit? Looks poor

Reply:
I am afraid I don't have any experience with that, I would have to look into it to make sure it is consistent across the entire page. If you want anything specific to change, let me know. I will take a look some other time.

Reply:
I think its about using some defualt CSS/config on entire docs. I'd just use Saleor font and color palette to be consistent 

cc @mirekm ;)

Reply:
Should be possible to do that: https://mermaid-js.github.io/mermaid/#/./theming

Dont make my mistake again and don't forget about the dark mode :D

Reply:
After we complete the docs I could fight with the styles